File: an/Intent/app/src/main/java/com/example/intent/MainActivity.java
package com.example.intent;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.media.Image;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;

public class MainActivity extends AppCompatActivity {

    View.OnClickListener listener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            Intent intent = new Intent();
            switch (v.getId()){
                case R.id.dial:
                    intent.setAction(intent.ACTION_DIAL);
                    intent.setData(Uri.parse("tel:10086"));
                    startActivity(intent);
                    break;
                case R.id.email:
                    intent.setAction(intent.ACTION_SENDTO);
                    intent.setData(Uri.parse("smsto:+8613382140617"));
                    intent.putExtra("sms_body", "Welcome to Microsoft");
                    startActivity(intent);
            }
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageButton dialButton = (ImageButton) findViewById(R.id.dial);
        ImageButton emailButton = (ImageButton) findViewById(R.id.email);

        dialButton.setOnClickListener(listener);
        emailButton.setOnClickListener(listener);
    }
}

File: an/Intent2/app/src/main/java/com/example/intent/MainActivity.kt
package com.example.intent

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ImageButton

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val imageButton = findViewById<ImageButton>(R.id.imageButton_phone)
        val imageButton1 = findViewById<ImageButton>(R.id.imageButton_sms)
        imageButton.setOnClickListener(listener)
        imageButton1.setOnClickListener(listener)
    }

    val listener = View.OnClickListener { view ->
        when (view.id) {
            R.id.imageButton_phone -> {val intent = Intent(Intent.ACTION_DIAL); intent.setData(Uri.parse("tel:10086"));
            startActivity(intent)}
            else -> {val intent = Intent(Intent.ACTION_SENDTO); intent.setData(Uri.parse("smsto:10086"));
            intent.putExtra("ms_body", "Hello Android"); startActivity(intent)}
        }
    }
}
